src/Offcanvas.php

```diff
@@ -4,235 +4,544 @@
+    /**
+     * Adds one or more CSS classes to the existing classes of the offcanvas component.
+     *
+     * Multiple classes can be added by passing them as separate arguments. `null` values are filtered out
+     * automatically.
+     *
+     * @param BackedEnum|string|null ...$class One or more CSS class names to add. Pass `null` to skip adding a class.
+     * For example:
+     *
+     * ```php
+     * $carousel->addClass('custom-class', null, 'another-class');
+     * ```
+     *
+     * @return self A new instance with the specified CSS classes added to existing ones.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     */
+    public function addClass(BackedEnum|string|null ...$class): self
     {
-        // TODO: fix the method call, there's no suffix anymore.
-        return $this->options['id'] ?? parent::getId($suffix);
+        $new = clone $this;
+        $new->cssClasses = [...$this->cssClasses, ...$class];
+
+        return $new;
     }
 
-    protected function toggleComponent(): string
+    /**
+     * Adds a CSS style for the offcanvas component.
+     *
+     * @param array|string $style The CSS style for the offcanvas component. If an array, the values will be separated by
+     * a space. If a string, it will be added as is. For example, `color: red`. If the value is an array, the values
+     * will be separated by a space. e.g., `['color' => 'red', 'font-weight' => 'bold']` will be rendered as
+     * `color: red; font-weight: bold;`.
+     * @param bool $overwrite Whether to overwrite existing styles with the same name. If `false`, the new value will be
+     * appended to the existing one.
+     *
+     * @return self A new instance with the specified CSS style value added.
+     */
+    public function addCssStyle(array|string $style, bool $overwrite = true): self
     {
-        return 'offcanvas';
+        $new = clone $this;
+        Html::addCssStyle($new->attributes, $style, $overwrite);
+
+        return $new;
     }
 
     /**
-     * Enable/disable body scroll when offcanvas show
+     * Sets the HTML attributes for the offcanvas component.
      *
-     * @link https://getbootstrap.com/docs/5.1/components/offcanvas/#backdrop
+     * @param array $attributes Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function scroll(bool $enabled = true): self
+    public function attributes(array $attributes): self
     {
         $new = clone $this;
-        $new->scroll = $enabled;
+        $new->attributes = $attributes;
 
         return $new;
     }
 
     /**
-     * Enable/disable offcanvas backdrop
+     * Sets whether the offcanvas component has a backdrop.
      *
-     * @link https://getbootstrap.com/docs/5.1/components/offcanvas/#backdrop
```

Comment:
A link is useful since there's an example of what backdrop is.

Reply:
done.